dali/python/nvidia/dali/pipeline.py

```diff
@@ -1203,6 +1257,22 @@ def iter_setup(self):
         data from NumPy arrays."""
         pass
 
+    def _generate_build_args(self):
+        ret = []
+        num_outputs = len(self._names_and_devices)
+        dtype = [self._output_dtype] * num_outputs if type(
+            self._output_dtype) is not list else self._output_dtype
+        ndim = [self._output_ndim] * num_outputs if type(self._output_ndim) is not list else self._output_ndim
+        if not (len(dtype) == len(ndim) == num_outputs):
+            raise RuntimeError(
+                f"Inconsistent output description. Length of provided output descriptions do not match. \n"
+                f"Expected num_outputs={num_outputs}.\nReceived:\noutput_dtype={dtype}\noutput_ndim={ndim}")
+
+        for (name, dev), dt, nd in zip(self._names_and_devices, dtype, ndim):
```

Comment:
nitpick: You could go with list comprehension here instead of ret = [] and append to be more pythonic.

Reply:
Done